website/docs/r/key_vault_managed_hardware_security_module.html.markdown

```diff
@@ -0,0 +1,85 @@
+---
+subcategory: "Key Vault"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_key_vault_managed_hardware_security_module"
+description: |-
+  Manages a Key Vault Managed Hardware Security Module.
+---
+
+# azurerm_key_vault_managed_hardware_security_module
+
+Manages a Key Vault Managed Hardware Security Module.
+
+## Example Usage
+
+```hcl
+data "azurerm_client_config" "current" {
+}
+
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_key_vault_managed_hardware_security_module" "example" {
+  name                       = "exampleKVHsm"
+  resource_group_name        = azurerm_resource_group.example.name
+  location                   = azurerm_resource_group.example.location
+  sku_name                   = "Standard_B1"
+  purge_protection_enabled   = false
+  soft_delete_retention_days = 90
+  tenant_id                  = data.azurerm_client_config.current.tenant_id
+  admin_object_ids           = [data.azurerm_client_config.current.object_id]
+
+  tags = {
+    Env = "Test"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of the Key Vault Managed Hardware Security Module. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the resource group in which to create the Key Vault Managed Hardware Security Module. Changing this forces a new resource to be created.
+
+* `location` - (Required) Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created.
+
+* `sku_name` - (Required) The Name of the SKU used for this Key Vault Managed Hardware Security Module. Possible value is `Standard_B1`.
+
+* `admin_object_ids` - (Required) Specifies a list of administrators object IDs for the key vault Managed Hardware Security Module.
+
+* `tenant_id` - (Required) The Azure Active Directory tenant ID that should be used for authenticating requests to the key vault Managed Hardware Security Module.
+
+* `purge_protection_enabled` - (Optional) Is Purge Protection enabled for this Key Vault Managed Hardware Security Module? Defaults to `false`. Changing this forces a new resource to be created.
+
+* `soft_delete_retention_days` - (Optional) The number of days that items should be retained for once soft-deleted. This value can be between `7` and `90` days. Defaults to `90`. Changing this forces a new resource to be created.
```

Comment:
this isn't defaulted in the schema - and from memory Terraform intentionally defaults to `7` for the other resource, so we should match that here

Reply:
for the key vault resource, it's also default 90 https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/keyvault/key_vault_resource.go#L196